### PR TITLE
EWPP-3102: Allow 2 new icons to be used in field_oe_icon of paragraphs.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_theme_helper/oe_theme_helper.services.yml [hunk 1]
```diff
@@ -36,6 +36,10 @@ services:
     arguments: ['@current_route_match', '@entity_type.manager', '@entity.repository']
     tags:
       - { name: event_subscriber }
+  oe_theme_helper.icon_options_subscriber:
+    class: Drupal\oe_theme_helper\EventSubscriber\IconOptionsEventSubscriber
+    tags:
+      - { name: event_subscriber }
```
Comment (sergepavle) [inline modules/oe_theme_helper/oe_theme_helper.services.yml:42]: Sorry, but I didn't get why event subscriber is implemented on oe_theme side again. As I see it is already done in oe_paragraph. Should it be updated only in oe_paragraph?

Comment (22Alexandra) [reply modules/oe_theme_helper/oe_theme_helper.services.yml:42]: I replied in the ticket.

Comment (sergepavle) [reply modules/oe_theme_helper/oe_theme_helper.services.yml:42]: Thanks. Looks good to me.

File: modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  /**
+   * Gets the icon options.
+   *
+   * @param \Drupal\oe_paragraphs\Event\IconOptionsEvent $event
+   *   Allowed format event object.
+   */
+  public function getIconOptions(IconOptionsEvent $event): void {
+    $event->setIconOptions([
```
Comment (sergepavle) [inline modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php]: But does it make sense to override all icon options in oe_theme? Maybe it is better to first use $event->getIconOptions(), update the received array, and $event->setIconOptions().

**Review by sergepavle — DISMISSED** [review]

File: modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    return [
+      IconOptionsEvent::class => 'getIconOptions',
+    ];
+  }
+
+  /**
+   * Gets the icon options.
+   *
+   * @param \Drupal\oe_paragraphs\Event\IconOptionsEvent $event
+   *   Allowed format event object.
```
Comment (nagyad) [inline modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php]: This is not correct.

File: modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+   */
+  public function getIconOptions(IconOptionsEvent $event): void {
```
Comment (nagyad) [inline modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php]: It's confusing for me to call this function the same as the event's function. We are rather altering the options we get from the event and re-setting it with the additional ones. `alterIconOptions`

File: modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      IconOptionsEvent::class => 'getIconOptions',
+    ];
+  }
+
+  /**
+   * Gets the icon options.
```
Comment (nagyad) [inline modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php]: `Alter the list of allowed paragraph icons. `

File: modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\EventSubscriber;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\oe_paragraphs\Event\IconOptionsEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Provides a list of allowed icons.
```
Comment (nagyad) [inline modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php]: `Event subscriber to alter the list of allowed paragraph icons.`

**Review by nagyad — CHANGES_REQUESTED** [review]

File: modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+class IconOptionsEventSubscriber implements EventSubscriberInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      IconOptionsEvent::class => 'getIconOptions',
```
Comment (nagyad) [inline modules/oe_theme_helper/src/EventSubscriber/IconOptionsEventSubscriber.php]: You forgot to change this to `alterIconOptions`

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

**Review by nagyad — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]

**Pull request merged**